test/spec/Radicle/Tests.hs

```diff
@@ -717,21 +717,48 @@ test_cbor =
             counterexample info $ got == expected
 
 
--- Tests all radicle files 'repl' dir. These should use the 'should-be'
--- function to ensure they are in the proper format.
+-- Tests radicle files. These should use the 'should-be' function to ensure
+-- they are in the proper format.
+-- Note that loaded files will also be tested, so there's no need to test files
+-- loaded by the prelude individually.
 test_source_files :: IO TestTree
-test_source_files = testGroup "Radicle source file tests" <$> do
-    (dir, files) <- sourceFiles
-    let radFiles = filter (\x -> ".rad" `isSuffixOf` x && "repl." `isInfixOf` x) files
-    sequence $ radFiles <&> \file -> do
+test_source_files = testGroup "Radicle source file tests" <$>
+    testOne "rad/prelude.rad"
+  where
+    testOne :: FilePath -> IO [TestTree]
+    testOne file = do
+        (dir, files) <- sourceFiles
+        allFiles <- forM files $ \f -> do
+            contents <- readFile (dir <> f)
+            pure (toS f, contents)
         contents <- readFile (dir <> file)
-        let (_, out) = runTestWith replBindings [] contents
+        let (r, out) = runTestWithFiles testBindings [] (fromList allFiles) contents
         let makeTest line = let (name, result) = T.span (/= '\'')
                                                $ T.drop 1
                                                $ T.dropWhile (/= '\'') line
                             in testCase (toS name) $ result @?= "' succeeded\""
-        pure $ testGroup file
-            $ [ makeTest ln | ln <- out, "\"Test" `T.isPrefixOf` ln ]
+        pure $ [testGroup file
+            $ testCase "doesn't throw" (isRight r @?= True)
+            : [ makeTest ln | ln <- out, "\"Test" `T.isPrefixOf` ln ]]
+
+test_macros :: [TestTree]
+test_macros =
+    [ testCase ":enter-chain keeps old bindings" $ do
+        let input = [ "(def x 0)"
+                    , "(:enter-chain \"blah\")"
+                    , "(def x 1)"
+                    , ":quit"
+                    , "x"
+                    ]
+            output = [ "0.0" ]
+        (_, result) <- runInRepl input
+        result @==> output
+    ]
+  where
+    -- In addition to the output of the lines tested, 'should-be's get
+    -- printed, so we take only the last few output lines.
+    r @==> out = reverse (take (length out) $ reverse r) @?= out
```

Comment:
You can use [`Data.List.isSuffixOf`](http://hackage.haskell.org/package/base-4.12.0.0/docs/Data-List.html#v:isSuffixOf)

Reply:
But then I'd need to manually print out the expected and got value rather than getting the automated error, and probably do this dance anyhow.